repo/blob/azure/azure_storage.go

```diff
@@ -112,13 +122,24 @@ func (az *azStorage) PutBlob(ctx context.Context, b blob.ID, data blob.Bytes, op
 
 	bc := az.bucket.NewBlockBlobClient(az.getObjectNameString(b))
 
-	_, err := bc.Upload(ctx, data.Reader(), &azblob.UploadBlockBlobOptions{})
+	ubo := &azblob.UploadBlockBlobOptions{
+		Metadata: timestampmeta.ToMap(opts.SetModTime, timeMapKey),
+	}
 
-	return translateError(err)
-}
+	resp, err := bc.Upload(ctx, data.Reader(), ubo)
+	if err != nil {
+		return translateError(err)
+	}
+
+	if opts.GetModTime != nil {
+		if opts.SetModTime.IsZero() {
+			*opts.GetModTime = *resp.LastModified
```

Comment:
wouldn't it be better to rely on `resp.LastModified` always whether or not `opts.SetModTime` was defined ?

Reply:
done, applied this everywhere